tendrl/commons/message.py

```diff
@@ -0,0 +1,47 @@
+import datetime
+from inspect import getframeinfo
+from inspect import stack
+import uuid
+
+
+class Message(object):
+    def __init__(self, message):
+        try:
+            if "message_id" not in message:
+                self.message_id = str(uuid.uuid4())
+            else:
+                self.message_id = message["message_id"]
+            self.timestamp = message["timestamp"]
+            self.priority = message["priority"]
+            self.publisher = message["publisher"]
+            self.payload = message["payload"]
+            # From which function, line and file error raised
+            if "caller" not in message:
+                caller = getframeinfo(stack()[1][0])
+                self.caller = {"filename": (caller.filename.rsplit('/', 1))[1],
+                               "line_no": caller.lineno,
+                               "function": caller.function}
+            else:
+                self.caller = message["caller"]
+            if not self.validate():
+                self.timestamp =  datetime.datetime.now().isoformat()
+                self.priority = "debug"
+                self.publisher =  "node-agent"
+                # Invalid obj does not have node_id
+                # because it is stored in syslog only
+                self.payload = {"message": message}
+
+        except KeyError as ex:
+            raise ex
+
+    def validate(self):
+        if (self.priority == "" or
+            self.timestamp == "" or
+            self.publisher == "" or
+            "message" not in self.payload):
+            return False
+        if "job_id" in self.payload:
+            if ("flow_id" not in self.payload or
+                "cluster_id" not in self.payload):
+                return False
```

Comment:
@brainfunked @shtripat @nthomas-redhat @nnDarshan @anmolbabu 
For job cluster_id is mandatory or not? because node related jobs have cluster id?

import cluster job does not have cluster id right?